### DISCUSSION TITLE
support for Windows neovim?

### DISCUSSION THREAD

**Discussion by mgan2014** [huggingface/llm.nvim#78] [Q&A]
I really like this plugin. It works well with the neovim in Ubuntu. For Windows, I can install this plugin. However, when the neovim starts, there is a warning "arch or OS not supported for x86_64 Windows_NT", something like tat. After a little bit research, I found the code in the lua folder and it seems that Windows is not supported. Is this correct? Thank you.

**Comment by life00** [comment]
I think it should work as long as you get llm-ls working. Make sure to specify the binary path (see readme).

**Reply by mgan2014** [reply]
Thank you for the reply. With your pointing me to the direction of the binary path setting, I added this to config.lua (in the folder AppData\Local\nvim-data\lazy\llm.nvim\lua\llm), seperately, as I installed llm-ls with Mason. I am not sure which one works so I tried both one at a time. Neither works.
    bin_path = vim.api.nvim_call_function("stdpath", { "data" }) .. "/mason/bin/llm-ls",
    bin_path = vim.api.nvim_call_function("stdpath", { "data" }) .. "/mason/bin",

I still get the same error as shown below.
![image](https://github.com/huggingface/llm.nvim/assets/8353442/43d70316-9c2c-48fb-b70a-f35a9afac4a9)

Any thoughts? BTW, I did confirm there is a windows executable called llm-ls under mason/bin/, so Mason did installed it correctly.

Thank you very much.

**Comment by mgan2014** [comment]
![image](https://github.com/huggingface/llm.nvim/assets/8353442/c2279653-dade-4a7e-a773-163f6aea6cd5)
here is the screenshot of the lsp config in the aforementioend config.lua file.

**Comment by life00** [comment]
So, I am not entirely sure how everything is stored in windows. If you use mason then that might be right and I guess mason should install the right version. I also faced this problem because I use aarch64 and the plugin doesn't automatically support it. Here is my lazy config that fixed it:

```lua
  {
    "huggingface/llm.nvim",
    opts = {
      backend = "ollama",
      model = "codellama:7b",
      accept_keymap = "<S-CR>",
      dismiss_keymap = "<CR>",
      url = "http://localhost:11434/api/generate",
      request_body = {
        options = {
          temperature = 0.2,
          top_p = 0.95,
        },
      },
      enable_suggestions_on_startup = false,
      lsp = {
        bin_path = vim.api.nvim_call_function("stdpath", { "data" }) .. "/mason/bin/llm-ls",
      },
    },
  },
```

If it doesn't work then try to find the binary manually and put the direct hard coded path.

**Reply by mgan2014** [reply]
Thanks again for your advices.
Finally, I got it working, with brutal force. Essentially, I go into language_server.lua and force the bin_path by replacing the download_llm_ls() function to point to llm-ls.exe directly. There may be other more elegant way to handle this, but I am happy with that the autosuggestion finally works for me! Bow to you!
<img width="973" alt="image" src="https://github.com/huggingface/llm.nvim/assets/8353442/b6ade532-1adf-4524-9b5e-42f8d32df504">

**Reply by mgan2014** [reply]
With one Mason Update, all the forced changes are wiped off. So, I went back to the script you posted and modified as below for the installation with Mason. Finally, it works without going to deep in lazy/llm.nvim/lua to make the changes over there.

lsp = {
        bin_path = vim.api.nvim_call_function("stdpath", { "data" }) .. "\\mason\\packages\\llm-ls\\llm-ls.exe",
      },

**Reply by McPatate** [reply]
Does this work @mgan2014? It is possible that we don't compile `llm-ls` binaries for your platform, but not sure.
